### ISSUE TITLE
How to set a PDO statement attribute?

### ISSUE THREAD

**Issue by thejahweh** [yiisoft/yii2#9967]
I have to use the each method to select in batches. The problem is, that I have to delete some columns in this foreach, which generates a table lock. Now I have to set the cursor `PDO::ATTR_CURSOR` to forward only `PDO::CURSOR_FWDONLY`. Is this even possible by now?

```php
$simpleQuery = SimpleObject::find();
// need something like: $simpleQuery->setPdoStmtAttr(\PDO::ATTR_CURSOR, \PDO::CURSOR_FWDONLY);
foreach ($simpleQuery->each() as $simpleObject) {
    // fancy things...
}
```

**Comment by samdark** [comment]
Yes. You can access PDO instance via connection class.

**Comment by thejahweh** [comment]
Yes but the `PDO::ATTR_CURSOR` attribute must be set on the pdo statement object.

**Comment by samdark** [comment]
`Command::$pdoStatement` is public.

**Comment by thejahweh** [comment]
And how do I use it in my snipped?

**Comment by samdark** [comment]
Same as you'd use it with PDO.

**Comment by thejahweh** [comment]
So it is not possible with the each statement?

**Comment by samdark** [comment]
Probably not. I know nothing about your use case.

**Comment by samdark** [comment]
Currently not.

**Comment by thejahweh** [comment]
Hmm I just looked into it, but I don't know where I should put the change in.

1.Loop the Attributes through the Query
```php
$query->setPdoAttributes([\PDO::ATTR_CURSOR => \PDO::CURSOR_FWDONLY])->each();
```
2.Make the method `Query::createCommand` return the same Command for every call, save it in a private/protected variable and just return it if called twice.
```php
$query->command->setPdoAttributes([\PDO::ATTR_CURSOR => \PDO::CURSOR_FWDONLY]);
$query->each();
```

**Comment by samdark** [comment]
`#1` looks better.
